spatialprofilingtoolbox/db/fractions_transcriber.py

```diff
@@ -19,76 +19,73 @@ def describe_fractions_feature_derivation_method():
     '''.lstrip().rstrip()
 
 
-def insert_new_data_analysis_study(database_config_file, study_name, specifier):
+def insert_new_data_analysis_study(database_connection_maker, study_name, specifier):
     timestring = str(datetime.datetime.now())
     name = f'{study_name} : {specifier} : {timestring}'
-    with DatabaseConnectionMaker(database_config_file) as dcm:
-        connection = dcm.get_connection()
-        cursor = connection.cursor()
-        cursor.execute('''
-        INSERT INTO data_analysis_study(name)
-        VALUES (%s) ;
-        INSERT INTO study_component(primary_study, component_study)
-        VALUES (%s, %s) ;
-        ''', (name, study_name, name))
-        cursor.close()
-        connection.commit()
+    connection = database_connection_maker.get_connection()
+    cursor = connection.cursor()
+    cursor.execute('''
+    INSERT INTO data_analysis_study(name)
+    VALUES (%s) ;
+    INSERT INTO study_component(primary_study, component_study)
+    VALUES (%s, %s) ;
+    ''', (name, study_name, name))
+    cursor.close()
+    connection.commit()
```

Comment:
Here you've unwrapped this statement from the `with` statement, whereas in other places you've newly wrapped statements into `with DatabaseConnectionMaker(` clauses. What's the difference between the two cases?

Reply:
I think the explanation is: `db.SchemaInfuser` for whatever reason was made to be a subclass of `DatabaseConnectionMaker` a long time ago, so that its methods can  more directly access the db I guess. And the step of computing and uploading the cell-fractions features (including the work of `transcribe_fraction_features` and indirectly the method you referenced) is a wrap up step for the view refreshing performed by `db.SchemaInfuser`. The `SchemaInfuser` just passes along its `self` to provide db access.
In general there seems to be a degree of freedom related to the granularity at which new database connections or cursors are created and used. In a multi-step process, one connection or one for each step?
There is no real strategy here except an aim for consistency. In the new code of this PR, there is a lot of consistency about this; when the cursor is created, which object carries it as temporary state, etc. In the older code the pattern is less clear.

Reply:
To elaborate a bit more: `DatabaseConnectionMaker` is only used if the connection is explicitly desired (e.g. to do commits at certain moments). Otherwise I'm preferred `DBCursor` which hides the connection and gives directly one cursor.